### DISCUSSION TITLE
Expose scopedResults to ais-state-results

### DISCUSSION THREAD

**Discussion by tkrugg** [algolia/instantsearch#5381] [Ideas]
**Feature ⚡️**

As a Vue InstantSearch user, I'd like to be able to access data of scopedResults from [ais-state-results](https://www.algolia.com/doc/api-reference/widgets/state-results/vue/).

### What is your use case for such a feature?

Say I have an InstantSearch instance with two indices side by side: `demo_ecommerce` and `demo_media`
I'd like to be able to show the total number of hits for my query, for the two indices.

`nbHits of demo_ecommerce + nbHits of demo_media`

[demo]( https://codesandbox.io/s/hungry-margulis-ngnzf?file=/src/App.vue )

Currently I can do this with [ais-autocomplete ](https://www.algolia.com/doc/api-reference/widgets/autocomplete/vue/ ) because it's one of the few widgets that's able to access `scopedResults` and exposes it in a slot. But I think it's more natural to do with `ais-state-results`.

### What is your proposal

1. move [connectStateResults](https://github.com/algolia/vue-instantsearch/blob/master/src/connectors/connectStateResults.js) to InstantSearch.js so it can live with other connectors and have equal chances to leverage new features and be tested, and even be used in Angular InstantSearch.
2. exposed `scopedResults`, maybe even `indices` similarly to  [connectAutocomplete](https://github.com/algolia/instantsearch.js/blob/638e437fdd80af0cfd38818f9da37a50f8f4343f/src/connectors/autocomplete/connectAutocomplete.ts) 
3. Document FAQ or pattern for accessing data from multiple indices at the same time.

**Comment by Haroenv** [comment]
one thing I'd like to change about state-results (in a major version or via a prop to enable it) is to actually _do_ return (empty) items in init and rendering the children using it, to prevent issues like https://github.com/algolia/vue-instantsearch/issues/935

see: 

https://github.com/algolia/vue-instantsearch/blob/513017a47d38ccddd0557c01ce732e6e5e7648d6/src/connectors/connectStateResults.js#L6-L16

https://github.com/algolia/vue-instantsearch/blob/513017a47d38ccddd0557c01ce732e6e5e7648d6/src/components/StateResults.vue#L4
